### PR TITLE
Add checkstyle and fix styling errors

### PR DESCRIPTION
**Pull request by jgwilson42**
Please, could you review this change, as we use the variable names from the source in the test cases that Deeptest generates it makes sense for the demo project to have better variable names in the source code.

### CODE REVIEW

File: src/main/java/com/diffblue/javademo/Search.java [hunk 1]
```diff
@@ -10,22 +10,25 @@
...
-  {
-    if("3b6e258214f894ab41c9deaaeb38d1fd9aeca9c7"
-         .equals(DigestUtils.sha1Hex(s)))
-    {
+  /**
+   * Does the sha1 of the input match a predefined item.
+   * @param input input string
+   * @return boolean
+   */
+
```
Comment (rjmunro) [inline src/main/java/com/diffblue/javademo/Search.java]: Stray newline here.

File: src/main/java/com/diffblue/javademo/Search.java [hunk 2]
```diff
@@ -10,22 +10,25 @@
...
     return found;
   }
 
-  public boolean isNeedleInHaystack(String s)
-  {
-    if("3b6e258214f894ab41c9deaaeb38d1fd9aeca9c7"
-         .equals(DigestUtils.sha1Hex(s)))
-    {
+  /**
+   * Does the sha1 of the input match a predefined item.
```
Comment (rjmunro) [inline src/main/java/com/diffblue/javademo/Search.java]: I'd say "Check if the sha1 of the input matches an expected value"

**Comment by jgwilson42** [comment]
@rjmunro thanks, I have made the two changes.

File: src/main/java/com/diffblue/javademo/Search.java [hunk 3]
```diff
@@ -10,22 +10,24 @@
...
-  {
-    if("3b6e258214f894ab41c9deaaeb38d1fd9aeca9c7"
-         .equals(DigestUtils.sha1Hex(s)))
-    {
+  /**
+   * Check if the sha1 of the input matches an expected value.
+   * @param input input string
+   * @return boolean
+   */
+  public boolean isNeedleInHaystack(String input) {
```
Comment (peterschrammel) [inline src/main/java/com/diffblue/javademo/Search.java]: You could rewrite it as:
```
public boolean checkPassword(String password) {
    String storedPasswordHash = "3b6e258214f894ab41c9deaaeb38d1fd9aeca9c7";
    if(storedPasswordHash.equals(DigestUtils.sha1Hex(password))) {
       return true;
     }
     return false;
}
```

File: src/main/java/com/diffblue/javademo/TicTacToe.java [hunk 4]
```diff
@@ -15,20 +15,20 @@
...
+  public int checkTicTacToePosition(int[] board) {
     // First check number of cells
-    if (a.length != 9) {
+    if (board.length != 9) {
       return -1;
     }
 
     // Next check number of moves each player made
     int diff = 0;
     for (int i = 0; i < 9; i++) {
```
Comment (peterschrammel) [inline src/main/java/com/diffblue/javademo/TicTacToe.java]: You could name these indices, `cell`, `row`, `column`.

**Pull request merged**